moler/connection_observer.py

```diff
@@ -159,6 +166,9 @@ def result(self):
         if self.cancelled():
             raise NoResultSinceCancelCalled(self)
         if self._exception:
+            if self._needed_exception_raise:
+                ConnectionObserver._change_exception_to_raised(self._exception, self._exception_time)
+                self._needed_exception_raise = False
```

Comment:
here we should just remove `self._exception` from `_not_reraised_exceptions`

Reply:
Done.